test/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2UnitTest.java

```diff
@@ -709,4 +710,39 @@ Response serve(IHTTPSession session) {
         //Then
         assertThat(httpMessagesSent, hasSize(greaterThan(0)));
     }
-}
+
+@Test
+public void shouldReportXssInJsonReponse() throws NullPointerException, IOException {
+    String test = "/shouldReportXssInJsonReponse/";
+    
+    this.nano.addHandler(new NanoServerHandler(test) {
+        @Override
+        Response serve(IHTTPSession session) {
+            String name = session.getParms().get("name");
+            String response;
+            if (name != null) {
+                response = getHtml("example.json",
+                        new String[][] {{"name", name}});
+            } else {
+                response = getHtml("NoInput.html");
+            }
+            Response resp = new Response(response);
+            resp.setMimeType("application/json");
+            return resp;
+        }
+    });
+    
+    HttpMessage msg = this.getHttpMessage(test + "?name=test");
+    
+    this.rule.init(msg, this.parent);
+
+    this.rule.scan();
+
+    assertThat(alertsRaised.size(), equalTo(1));
+    assertTrue(alertsRaised.get(0).getName().contains("JSON"));
```

Comment:
It's preferable to use `assertThat` (with `containsString`), it gives a more meaningful exception/error message.